### ISSUE TITLE
[TODO] [Day 4] [실험] v087 모듈화 이후 LB 0.3243 리그레션 재현·원인 규명·복구

### ISSUE THREAD

**Issue by iejob** [AIBootcamp14/computervisioncompetition-cv-1#25]
## 📊 실험 개요

**실험 이름:** v087-regression-debug
**담당자:** @iejob
**예상 소요시간:** 6–8h
**목표 성능:** Local CV 0.90+, LB 0.82+
**마감일:** 9/05 24:00

---

## 🎯 실험 목표

* [x] 모듈화 코드 기준 재학습 및 재추론
* [x] 리그레션 원인 규명(설정/데이터/변환/모델/추론)
* [x] 제출 파일 생성 및 점수 복구

---

### 전처리

* [x] `meta.csv` 라벨 맵 ↔ 모델 head 순서 일치 검증
* [x] 이미지 로더 RGB 고정 및 Normalize(mean/std) 일치화
* [x] 검증 변환 ↔ 추론 변환 완전 동등성 확인(Resize/Pad/Normalize 동일)

### 모델 설정

* [x] 백본/헤드/`num_classes=17`/`pooling`/`drop_rate`/`drop_path_rate` 동기화
* [x] `pretrained=True` 학습·추론 모두 강제(ckpt 로드 전 생성 파라미터 일치)
* [x] ckpt와 모델 head shape 정합성 점검

### 학습 설정

* [x] optimizer/scheduler/lr/epochs/batch/AMP/grad\_clip 동등화
* [x] `img_size` 학습·검증·추론 동일화
* [x] config 단일 소스(`configs/train.yaml`, `configs/infer.yaml`)로 관리

### 검증 전략

* [x] 5-Fold Stratified(seed=42, shuffle=True) 동일 분할 사용
* [x] 폴드별 best ckpt 저장 및 OOF 집계
* [x] OOF macro F1 ↔ LB 점수 비교

---

## 📈 성공 기준

* **필수 달성:**

* [x] 설정 동등성 리포트 생성(YAML diff/변환 diff/라벨맵 diff)

* [x] OOF macro F1 ≥ 0.90 재현

* [x] 제출 파일 형식 검사 통과(ID 정렬/NaN/중복 無)

* [x] 추론 로그에 실제 로드 ckpt 경로 기록 확인

---

## 🔧 기술적 세부사항

**개발 환경:**

* GPU: (프로젝트 환경에 맞춤)
* 프레임워크: PyTorch
* 로깅: `src/utils/logger.py`(stdout/stderr redirect)
* 산출물: `experiments/train/<date>/<run_id>/` / 로그: `logs/train|infer/`

---

## 📦 설치/실행 명령어 예시

```bash
# 1) 환경 점검
python -V && pip list | grep -E "torch|timm|albumentations|opencv|pandas|numpy|Pillow"

# 2) 설정 동등성 점검 (train.yaml vs infer.yaml, 학습/추론 로그 비교)
python -m tools.diff_config \
  --train-yaml configs/train.yaml \
  --infer-yaml configs/infer.yaml \
  --train-log logs/train/train_*.log \
  --infer-log logs/infer/infer_*.log

# 3) 라벨맵/헤드 정합성
python -m tools.check_labelmap \
  --meta data/raw/meta.csv \
  --ckpt experiments/train/20250904/v087/ckpt/best_fold0.pth

# 4) 변환 동등성 검사
python -m tools.check_transforms \
  --train-cfg configs/train.yaml \
  --infer-cfg configs/infer.yaml

# 5) 재학습 (1 fold 재현)
python -m src.training.train_main --config configs/train.yaml --valid_fold 0

# 6) 재추론 (명시 ckpt, 동일 img_size/정규화/tta)
python -m src.inference.infer_main \
  --config configs/infer.yaml \
  --ckpt experiments/train/20250904/v087/ckpt/best_fold0.pth \
  --out submissions/20250904/infer_v087_fix.csv
```

---

**참고 자료:**

* [x] Training Pipeline 실행 가이드
* [x] Inference Pipeline 실행 가이드
